### PR TITLE
openstack vagrant improvements

### PR DESCRIPTION
**Pull request by akostadinov**
Without VM prefixes it's very hard to share a cloud account like openstack because you can hardly know which instance is yours and which not. 6384f3d allows to set a VM name prefix with an environment variable

Without floating IP it's imposisble to use openstack vagrant setup in most cases because you need a floating IP to access instances. 188fb06 makes it possible to specify floating IPs. The most usable values are `:auto` for floating ip and `false` for pool. if `:auto` is not used then I can only see single instance setup possible. `false` means auto for pools. Counter inthuitive but this is an openstack bug. See https://github.com/cloudbau/vagrant-openstack-plugin/pull/109

Install `which` utility missing from fedora cloud image.

Update: I rebased as I forgot two other places that need `which` installation.

### CODE REVIEW

**Comment by smarterclayton** [comment]
@danmcp review

File: Vagrantfile [hunk 1]
```diff
@@ -244,7 +245,12 @@ Vagrant.configure(VAGRANTFILE_API_VERSION) do |config|
       os.image        = voc['image']          || creds['OSImage']    || /Fedora/         # Regex or String
       os.ssh_username = user = voc['ssh_user']|| creds['OSSshUser']  || "root"           # login for the VM instance
       os.server_name  = ENV['OS_HOSTNAME']    || vagrant_openshift_config['instance_name'] # name for the instance created
-      full_provision(override.vm, user)
+      floating_ip_pool = creds['OSFloatingIPPool'] || ENV['OS_FLOATING_IP_POOL']
```
Comment (danmcp) [inline Vagrantfile]: Do you mind collapsing these next 6 lines into 2?

File: Vagrantfile [hunk 2]
```diff
@@ -244,7 +245,12 @@ Vagrant.configure(VAGRANTFILE_API_VERSION) do |config|
       os.image        = voc['image']          || creds['OSImage']    || /Fedora/         # Regex or String
       os.ssh_username = user = voc['ssh_user']|| creds['OSSshUser']  || "root"           # login for the VM instance
       os.server_name  = ENV['OS_HOSTNAME']    || vagrant_openshift_config['instance_name'] # name for the instance created
-      full_provision(override.vm, user)
```
Comment (danmcp) [inline Vagrantfile]: It's not clear why this line is being removed.

Comment (akostadinov) [inline Vagrantfile]: not intended, perhaps my rebase this morning caused it, I'll fix, thank you

File: Vagrantfile (hunk 1 above)
Comment (akostadinov) [inline Vagrantfile]: Possible but would 2 two very long lines. We need to have `:auto` sym and `false` as value, not string to get desired behavior. If you confirm I can make it 2 lines but would be hardly readable. Like:
```
os.floating_ip = (creds['OSFloatingIPPool'] || ENV['OS_FLOATING_IP_POOL']) == ":auto" ? :auto : (creds['OSFloatingIPPool'] || ENV['OS_FLOATING_IP_POOL'])
```
What do you suggest?

File: Vagrantfile [hunk 3]
```diff
@@ -245,6 +246,12 @@ Vagrant.configure(VAGRANTFILE_API_VERSION) do |config|
       os.ssh_username = user = voc['ssh_user']|| creds['OSSshUser']  || "root"           # login for the VM instance
       os.server_name  = ENV['OS_HOSTNAME']    || vagrant_openshift_config['instance_name'] # name for the instance created
       full_provision(override.vm, user)
+      floating_ip_pool = creds['OSFloatingIPPool'] || ENV['OS_FLOATING_IP_POOL']
```
Comment (akostadinov) [inline Vagrantfile]: @danmcp said:
Do you mind collapsing these next 6 lines into 2?
@akostadinov said:
Possible but would 2 two very long lines. We need to have `:auto` as a sym and `false` as value, not string to get desired behavior. If you confirm I can make it 2 lines but would be hardly readable. Like:
```
os.floating_ip = (creds['OSFloatingIPPool'] || ENV['OS_FLOATING_IP_POOL']) == ":auto" ? :auto : (creds['OSFloatingIPPool'] || ENV['OS_FLOATING_IP_POOL'])
```
What  would you suggest?

Comment (danmcp) [inline Vagrantfile]: For floating_ip, do you really want it to be a sym in one case (:auto) and not in the other?  Meaning is this wrong:

os.floating_ip  = (creds['OSFloatingIP'] || ENV['OS_FLOATING_IP']).to_sym

2 lines is ok for floating_id_pool.

Comment (akostadinov) [inline Vagrantfile]: As far as I'm reading docs, `floating_ip` takes an IP or `:auto`. Using a static IP does not make sense in a cluster config but given cluster config is not possible with anything but virtualbox at the moment, I assume people are usually running a single instance to contain master and minion. So one could want to specify a static IP address in that property. And I don't think  something like `:"10.10.10.10"` makes sense. Also `nil.to_sym` in your example could be a problem. I'll update PR to have 2 lines less.

File: Vagrantfile [hunk 4]
```diff
@@ -245,6 +246,12 @@ Vagrant.configure(VAGRANTFILE_API_VERSION) do |config|
       os.ssh_username = user = voc['ssh_user']|| creds['OSSshUser']  || "root"           # login for the VM instance
       os.server_name  = ENV['OS_HOSTNAME']    || vagrant_openshift_config['instance_name'] # name for the instance created
       full_provision(override.vm, user)
+
+         # floating ip usually needed for acessing machines
```
Comment (danmcp) [inline Vagrantfile]: Formatting seems to be off here.  Mixing spaces and tabs?

**Comment by danmcp** [comment]
[merge]

**Pull request merged**